.github/workflows/release-docker.yml

```diff
@@ -0,0 +1,34 @@
+---
+name: Release Docker images
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    tags:
+      - 'v*'
+
+jobs:
+  plugin-compiler:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout of tyk
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 1
+
+      - uses: docker/setup-buildx-action@v2
+
+      - id: docker-tag
+        uses: yuya-takeyama/docker-tag-from-github-ref-action@v1
+
+      - name: Plugin compiler
+        uses: docker/build-push-action@v3
+        with:
+          push: false         # startsWith(github.ref, 'refs/tags')
+          context: "."
+          file: docker/plugin-compiler/5.1/Dockerfile
```

Comment:
I guess this still needs a mechanism to get parametrized based on the release branch, so it picks the right file?

Reply:
Yup, gromit or local templating, essentially a `branch: dockerfile` mapping